corehq/apps/geospatial/static/geospatial/js/case_grouping_map.js

```diff
@@ -572,6 +572,11 @@ hqDefine("geospatial/js/case_grouping_map",[
                 return;
             }
 
+            // Groups need to be unlocked to load new case data
+            if (groupLockModelInstance.groupsLocked()) {
+                groupLockModelInstance.toggleGroupLock();
```

Comment:
nit: can't we simply unlock directly instead of checking for it first?

Reply:
Since it is a toggle function, if it is the case that it's already unlocked we would end up incorrectly locking the grouping.

Reply:
Right. I didn't mean to say we should call this same function but add a new function to just unlock if its not present already and call that.

Reply:
To keep things DRY, we would need to implement something as follows:
```
        self.toggleGroupLock = function () {
            self.groupsLocked(!self.groupsLocked());
            handleNewGroupLockState();
        };

        self.setGroupLock = function (val) {
            self.groupsLocked(val);
            handleNewGroupLockState();
        }

        function handleNewGroupLockState () {
            if (self.groupsLocked()) {
                mapModel.mapInstance.scrollZoom.disable();
                setCaseGroups();
            } else {
                mapModel.mapInstance.scrollZoom.enable();
                clearCaseGroups();
            }
        }
```
Adding two new functions feels like we might be adding unnecessary additional complexity to get rid of the conditional. I'm inclined to leave the code as is, but happy to think through changing it if you believe it would make the code more readable.

Reply:
Ya fine to leave it as it is if you think the code refactor is not worth it.